File: backend/hostel-management/src/main/java/com/hostel/hostel_management/service/FeeService.java
package com.hostel.hostel_management.service;

import java.util.List;

import com.hostel.hostel_management.entity.Fee;

public interface FeeService {
    Fee createFee(Fee fee);
    List<Fee> getStudentFees(Long studentId);
    List<Fee> getDefaulters();
    Fee updateFee(Long feeId, Fee updatedFee);
    Fee getByFeeId(Long feeId);
    List<Fee> getAllFees();
}

File: frontend/hostel-frontend/src/services/feeService.js
import API from "../api/axios";

export const createFee = (data) =>
  API.post("/fees", data);

export const updateFee = (feeId,data) =>
  API.put(`/fees/${feeId}`,data);

export const getAllFees = () => {
  return API.get("/fees");
};

export const getFeeById = (feeId) =>
  API.get(`/fees/${feeId}`);

export const getStudentFees = (studentId) =>
  API.get(`/fees/student/${studentId}`);

export const getDefaulters = () =>
  API.get("/fees/defaulters");

export const makePayment = (feeId,data) =>
  API.post(`/fees/payments/${feeId}`,data);

export const getStudentPayments = (studentId) =>
  API.get(`/fees/payments/student/${studentId}`);

export const getReceipt = (paymentId) =>
  API.get(`/fees/payments/receipt/${paymentId}`);